### PR TITLE
test(link): refactor for cypress tests

### PR DESCRIPTION
**Pull request by DlgSHi**
### Proposed behaviour

Refactor Cypress tests for the `Link` component to use new cypress-component-react framework for testing.
Add `link.test.js`. 

### Current behaviour

Current process is the regular Cypress approach with BDD.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- [x] Run `npx cypress open --component` to check if there is newly added test.file
- [x] Check if the `link.test.js` file passed
- [x] Check all tests were moved/refactored from .feature file -> .test.js file
- [x] Check all unnecessary files were deleted, including .json file
- [x] Run `npx cypress run --component` to check none of the other *.test.js files have regressed
- [x] Run `npx cypress run --e2e` to check none of the feature files have regressed

### CODE REVIEW

**Review by divyajindel — APPROVED** [review]

File: src/components/link/link.test.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    >
+      <Link href="#foo" target="_blank" rel="noreferrer noopener" {...props}>
+        This is a link
+      </Link>
+    </div>
+  );
+};
+
+context("Test for Link component", () => {
+  describe.only("Check props for Link component", () => {
```
Comment (ZhuoyuJin) [inline src/components/link/link.test.js:33]: Do we need ".only" here? Also "Check" should be "check"

File: src/components/link/link.test.js [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+        link().should("have.attr", "class").and("contain", className);
+      }
+    );
+
+    it("should render Link disabled", () => {
+      CypressMountWithProviders(<LinkComponent disabled />);
+
+      link().should("have.attr", "disabled");
+      link().children().should("have.css", "color", "rgba(0, 0, 0, 0.3)");
```
Comment (ZhuoyuJin) [inline src/components/link/link.test.js:47]: Should we care about the background color here? Wouldn't "link().should("have.attr", "disabled");" be enough to test the "disabled" prop?

Comment (DlgSHi) [reply src/components/link/link.test.js:47]: Actually the color is changing for the `children`, so I think so :)

File: src/components/link/link.test.js (hunk 1 above)
Comment (DlgSHi) [reply src/components/link/link.test.js]: good spot

**Review by ZhuoyuJin — DISMISSED** [review]

**Review by divyajindel — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Pull request merged**